From 0ce382c275b087e866517c003e565f8cc4855bdd Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 29 Aug 2020 23:19:26 +0200
Subject: [PATCH] CMakeLists.txt: do not force SHARED

Allow the user to build static libraries by removing SHARED from
ADD_LIBRARY calls.

Here is an extract of
https://cmake.org/cmake/help/latest/command/add_library.html:

"If no type is given explicitly the type is STATIC or SHARED based on
whether the current value of the variable BUILD_SHARED_LIBS is ON."

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: probably not upstreamable as is because it will
conflict with https://github.com/redis/hiredis/pull/851. So, this patch
will probably have to be reworked for 1.0.x]
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1beccc6..fed79fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,7 @@ IF(WIN32)
     ADD_COMPILE_DEFINITIONS(_CRT_SECURE_NO_WARNINGS WIN32_LEAN_AND_MEAN)
 ENDIF()
 
-ADD_LIBRARY(hiredis SHARED ${hiredis_sources})
+ADD_LIBRARY(hiredis ${hiredis_sources})
 
 SET_TARGET_PROPERTIES(hiredis
     PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE
@@ -97,7 +97,7 @@ IF(ENABLE_SSL)
     FIND_PACKAGE(OpenSSL REQUIRED)
     SET(hiredis_ssl_sources
         ssl.c)
-    ADD_LIBRARY(hiredis_ssl SHARED
+    ADD_LIBRARY(hiredis_ssl
             ${hiredis_ssl_sources})
 
     IF (APPLE)
-- 
2.28.0

